### PR TITLE
Indirect take two

### PR DESCRIPTION
**Pull request by kalmarek**
Please test for `os=:Darwin` and `os=:Windows`.

Note that `using SCS` will fail if You have previous version installed with
```julia
LoadError: UndefVarError: direct not defined
```
This is because of precompilation and code generation in `c_wrapper.jl` I don't know how to get around it to tell user that running `Pkg.build("SCS")` is required.

### CODE REVIEW

File: deps/build.jl [hunk 1]
```diff
@@ -4,25 +4,25 @@ using BinDeps
 
 blasvendor = Base.BLAS.vendor()
 
+libs = Dict(:direct=>"libscsdir", :indirect=>"libscsindir")
```
Comment (mlubin) [inline deps/build.jl]: These names aren't going to change frequently, so just remove this extra level of indirection.

File: deps/build.jl [hunk 2]
```diff
@@ -4,25 +4,25 @@ using BinDeps
...
 end
 
-scs = library_dependency("scs", aliases=aliases)
+direct = library_dependency("direct", aliases=[libs[:direct]])
+indirect = library_dependency("indirect", aliases=[libs[:indirect]])
 
 if is_apple()
     using Homebrew
-    provides(Homebrew.HB, "scs", scs, os = :Darwin)
+    provides(Homebrew.HB, [libs[:direct], libs[:indirect]], [direct, indirect], os = :Darwin)
```
Comment (mlubin) [inline deps/build.jl]: Revert this change. Homebrew will still provide a single package called "scs".

File: deps/build.jl [hunk 3]
```diff
@@ -31,13 +31,13 @@ provides(Sources, URI("https://github.com/cvxgrp/scs/archive/v$version.tar.gz"),
 # CFLAGS="-DDLONG -DCOPYAMATRIX -DUSE_LAPACK -DCTRLC=1" LDFLAGS="-L$HOME/julia32/usr/bin -lopenblas" make CC=i686-w64-mingw32-gcc out/libscsdir.dll
 # mv out bin32
 provides(Binaries, URI("https://cache.julialang.org/https://bintray.com/artifact/download/tkelman/generic/scs-$win_version-r2.7z"),
-    [scs], unpacked_dir="bin$(Sys.WORD_SIZE)", os = :Windows,
+    [direct, indirect], unpacked_dir="bin$(Sys.WORD_SIZE)", os = :Windows,
```
Comment (mlubin) [inline deps/build.jl]: Just comment out this "provides" for windows and leave a TODO. That's better than leaving code that is known to not work.

File: src/c_wrapper.jl [hunk 4]
```diff
@@ -44,6 +44,13 @@ function SCS_solve(m::Int, n::Int, A::SCSVecOrMatOrSparse, b::Array{Float64},
         slack::Vector{Float64}=Float64[];
         options...)
 
+    T = SCS.Indirect # the default method
```
Comment (mlubin) [inline src/c_wrapper.jl]: At the level of the C wrapper, `Direct` and `Indirect` should be parameters to `SCS_solve` as below, not stuffed into `options`.

**Comment by mlubin** [comment]
> I don't know how to get around it to tell user that running Pkg.build("SCS") is required.

That's fine, because nobody will have SCS 2.0 installed anyway, and they'll be forced to rebuild.

File: deps/build.jl (hunk 2 above)
Comment (kalmarek) [reply deps/build.jl]: Do You mean
```julia
provides(Homebrew.HB, "scs", [direct, indirect], os = :Darwin)
```
so that Homebrew "scs" package will provide both direct and indirect lib?

Comment (mlubin) [reply deps/build.jl]: Yes. But probably best to comment out this code also and leave a TODO because it doesn't work.

File: src/c_wrapper.jl (hunk 4 above)
Comment (kalmarek) [reply src/c_wrapper.jl]: I moved this default and options "cleaning" to `optimize!` in `SCSSolverInterface.jl`

File: deps/build.jl [hunk 5]
```diff
@@ -64,13 +59,21 @@ ENV2["CFLAGS"] = cflags
...
+        GetSources(direct)
         CreateDirectory(joinpath(prefix, "lib"))
-        FileRule(joinpath(prefix, "lib", libname), @build_steps begin
+
+        FileRule(joinpath(prefix, "lib", "libscsdir.$(Libdl.dlext)"), @build_steps begin
             ChangeDirectory(srcdir)
-            setenv(`make BLASLDFLAGS= out/$libname`, ENV2)
-            `mv out/$libname $prefix/lib`
+            setenv(`make BLASLDFLAGS= out/libscsdir.$(Libdl.dlext)`, ENV2)
+            `mv out/libscsdir.so $prefix/lib`
```
Comment (mlubin) [inline deps/build.jl]: The build is failing on OS X because of the `.so` extension here: https://travis-ci.org/JuliaOpt/SCS.jl/jobs/313571982#L360
Please try to get OS X passing.

**Comment by mlubin** [comment]
Thanks!
